core/vibe/core/drivers/libasync.d

```diff
@@ -1315,6 +1329,7 @@ final class LibasyncTCPConnection : TCPConnection/*, Buffered*/ {
 	}
 
 	void acquireReader() {
+		assert(!m_settings.writer.isWaiting, "Acquiring reader that is already in use.");
```

Comment:
I have almost no knowledge of the code, but shouldn't this be `m_settings.reader.isWaiting`?